app.py

```diff
@@ -1568,11 +1568,13 @@ def save_program(user):
     # For now, we bring all existing programs for the user and then search within them for repeated names.
     programs = DATABASE.programs_for_user(user['username']).records
     program_id = uuid.uuid4().hex
+    program_public = None
     overwrite = False
     for program in programs:
         if program['name'] == body['name']:
             overwrite = True
             program_id = program['id']
+            program_public = program.get('public', None)
```

Comment:
Nitpicky; do we perhaps want to default to `False`?

Reply:
I choose to set it to `None` because this is how we currently handle public programs. The None value is simply not stored in the db (so `public` doesn't exist). I agree that `False` might be better, but this way it's consistent with our current implementation and we make sure that we don't break anything else. (We might simply check if the `public` value exists somewhere instead of a True/False value).

Reply:
Aha, I see! Makes sense :)